### PR TITLE
docs: add method docs

### PR DESCRIPTION
**Pull request by dtdang**
### What I did

Add files necessary to generate method docs for ape-alchemy plugin
#24 

### How I did it
Added missing doc strings that were not already defined in ape core. 
### How to verify it
Verify the docs make sense that were added
### Checklist

- [ ] Passes all linting checks (pre-commit and CI jobs)
- [ ] New test cases have been added and are passing
- [ ] Documentation has been updated
- [ ] PR title follows [Conventional Commit](https://www.conventionalcommits.org/en/v1.0.0/) standard (will be automatically included in the changelog)

### CODE REVIEW

File: .github/workflows/docs.yaml [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+                git add .
+                git commit -m "Update documentation" -a || true
+              fi
+        - name: Push changes
+          uses: ad-m/github-push-action@master
+          if: ${{ github.event_name != 'pull_request' }}
+          with:
+              branch: gh-pages
+              directory: gh-pages
+              github_token: ${{ secrets.GITHUB_TOKEN }}
```
Comment (unparalleled-js) [inline .github/workflows/docs.yaml:56]: Missing new line at end of file

File: ape_alchemy/providers.py [hunk 2]
```diff
@@ -32,6 +32,10 @@ class AlchemyFeatureNotAvailable(AlchemyProviderError):
 
 
 class MissingProjectKeyError(AlchemyProviderError):
+    """
+    An error raised when not using one of the options available.
```
Comment (unparalleled-js) [inline ape_alchemy/providers.py:36]:
```suggestion
    An error raised when there is no API key set.
```

File: ape_alchemy/providers.py [hunk 3]
```diff
@@ -48,6 +52,15 @@ class AlchemyEthereumProvider(Web3Provider, UpstreamProvider):
 
     @property
     def uri(self):
+        """
+        Build from `ecosystem` and `network`.
```
Comment (unparalleled-js) [inline ape_alchemy/providers.py:56]:
```suggestion
        Your Alchemy RPC URI, including the project ID.
```

File: docs/index.md [hunk 4]
```diff
@@ -0,0 +1,17 @@
+# Ape-Docs
```
Comment (unparalleled-js) [inline docs/index.md:1]:
```suggestion
# Ape-Alchemy Documentation

```

File: ape_alchemy/providers.py [hunk 5]
```diff
@@ -32,6 +32,10 @@ class AlchemyFeatureNotAvailable(AlchemyProviderError):
 
 
 class MissingProjectKeyError(AlchemyProviderError):
```
Comment (unparalleled-js) [inline ape_alchemy/providers.py:34]: We could consider moving this to an `exceptions.py` file to be consistent.

**Review by unparalleled-js — CHANGES_REQUESTED** [review]
Looking great!
Here is some feedback.

File: ape_alchemy/providers.py (hunk 5 above)
Comment (dtdang) [reply ape_alchemy/providers.py:34]: would I move all 3 errors to the `exceptions.py` file?

Comment (unparalleled-js) [reply ape_alchemy/providers.py:34]: Yes, if we make a `exceptions.py`, let's put all of them in there

File: .github/workflows/docs.yaml [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+
```
Comment (unparalleled-js) [inline .github/workflows/docs.yaml:57]:
```suggestion

```

File: ape_alchemy/providers.py [hunk 7]
```diff
@@ -47,6 +30,15 @@ class AlchemyEthereumProvider(Web3Provider, UpstreamProvider):
 
     @property
     def uri(self):
+        """
+         Your Alchemy RPC URI, including the project ID.
+
+        `ecosystem`: choose from supported ecosystems
+        `network`: choose a network to connect to
+
+        Returns:
```
Comment (unparalleled-js) [inline ape_alchemy/providers.py:39]: `Returns:` / `Args:` statements are not needed for properties

**Review by unparalleled-js — APPROVED** [review]
Couple small things but looks great

File: .github/workflows/docs.yaml (hunk 6 above)
Comment (NotPeopling2day) [reply .github/workflows/docs.yaml]: Updated [3f4682c](https://github.com/ApeWorX/ape-alchemy/pull/23/commits/3f4682cdab7724cc3bc6f70a1c6a45ae5e91da74)

File: ape_alchemy/providers.py (hunk 7 above)
Comment (NotPeopling2day) [reply ape_alchemy/providers.py]: Updated [3f4682c](https://github.com/ApeWorX/ape-alchemy/pull/23/commits/3f4682cdab7724cc3bc6f70a1c6a45ae5e91da74)

**Pull request merged**